### PR TITLE
DOC: improve pd.Timestamp docs with more examples and clarify differe…

### PR DESCRIPTION
**Pull request by mortada**
…nt forms of API usage

### CODE REVIEW

**Review by gfyoung — APPROVED** [review]

File: pandas/_libs/tslibs/timestamps.pyx [hunk 1]
```diff
@@ -380,10 +380,11 @@ class Timestamp(_Timestamp):
         Value to be converted to Timestamp
     freq : str, DateOffset
         Offset which Timestamp will have
-    tz : string, pytz.timezone, dateutil.tz.tzfile or None
+    tz : str, pytz.timezone, dateutil.tz.tzfile or None
         Time zone for time which Timestamp will have.
-    unit : string
-        numpy unit used for conversion, if ts_input is int or float
+    unit : str
+        Unit used for conversion if ts_input is of type int or float.
```
Comment (jreback) [inline pandas/_libs/tslibs/timestamps.pyx]: let's add the valid values (D, h, m, s, ms us, ns)

File: pandas/_libs/tslibs/timestamps.pyx [hunk 2]
```diff
@@ -405,9 +406,19 @@ class Timestamp(_Timestamp):
 
     Examples
     --------
+    Using the primary calling convention:
+
     >>> pd.Timestamp('2017-01-01T12')
     Timestamp('2017-01-01 12:00:00')
```
Comment (jreback) [inline pandas/_libs/tslibs/timestamps.pyx]: can you add some commentary before each of these examples

**Review by jreback — CHANGES_REQUESTED** [review]
thanks, some comments.

**Comment by mortada** [comment]
@jreback updated, please take a look

File: pandas/_libs/tslibs/timestamps.pyx [hunk 3]
```diff
@@ -405,9 +407,23 @@ class Timestamp(_Timestamp):
...
+    This converts a float representing a Unix epoch in units of seconds
+    >>> pd.Timestamp(1513393355.5, unit='s')
+    Timestamp('2017-12-16 03:02:35.500000')
+
+    This converts an int representing a Unix-epoch in units of seconds
+    and for a particular timezone
+    >>> pd.Timestamp(1513393355, unit='s', tz='US/Pacific')
+    Timestamp('2017-12-15 19:02:35-0800', tz='US/Pacific')
+
+    Using the other two forms that minic the API for ``datetime.datetime``:
```
Comment (jreback) [inline pandas/_libs/tslibs/timestamps.pyx]: mimic

File: pandas/_libs/tslibs/timestamps.pyx [hunk 4]
```diff
@@ -405,9 +407,23 @@ class Timestamp(_Timestamp):
 
     Examples
     --------
+    Using the primary calling convention:
```
Comment (jreback) [inline pandas/_libs/tslibs/timestamps.pyx]: do we use this nomenclature above? if not can you reconcile

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_libs/tslibs/timestamps.pyx (hunk 3 above)
Comment (mortada) [reply pandas/_libs/tslibs/timestamps.pyx]: good catch, will fix

File: pandas/_libs/tslibs/timestamps.pyx (hunk 4 above)
Comment (mortada) [reply pandas/_libs/tslibs/timestamps.pyx]: yes in the above the doc says:
```
    Notes
    -----
    There are essentially three calling conventions for the constructor. The
    primary form accepts four parameters. They can be passed by position or
    keyword.

    The other two forms mimic the parameters from ``datetime.datetime``. They
    can be passed by either position or keyword, but not both mixed together.
```

**Comment by mortada** [comment]
@jreback PR is updated

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**